### PR TITLE
[react] Add `media` attribute support for `meta`

### PR DESCRIPTION
**Pull request by brc-dd**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - <https://github.com/whatwg/html/pull/6569>
  - <https://html.spec.whatwg.org/multipage/semantics.html#the-meta-element> 
  - <https://html.spec.whatwg.org/multipage/semantics.html#attr-meta-media>
  - <https://html.spec.whatwg.org/multipage/semantics.html#meta-theme-color>
  -  <https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta/name/theme-color>
  - <https://web.dev/add-manifest/#theme-color>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

[Playground Link](https://www.typescriptlang.org/play?&dtPR=54936&install-plugin=playground-dt-review#code/PTAEAsBdIBwZwFwgCYFMBuqA2B7GqAnAOgFscAvASyywEMicCBzYVAOwFoBVAZWGRwBjOMADqqAEbAAEgBUAsgBlgAUSyoS7SME2RawNrU3BI4Dag6CcuAgChblEjEaRQAJVS1BrgGYEcJKAA5ASe3kEA3PZWbHCu0p7ICO5hkEQAYgDCoAC8oAAUAJS5AHwFtqCgADxmtMglFZXVurSghpo5AESm5pbWjJ2gmsiUtF35MKE+hHB9NhxwgmaayViUTFCFgzGQWl0A7uCUu4PADU3NqHptRqhdPZpzA0OoI2OdE1MzTwQLS+bJZC0AgAay2oB2e06EjoghBp3O1WAtXqtkKUVsqAAHs4CK40D5aABXLDxRJRIA)

### CODE REVIEW

**Comment by brc-dd** [comment]
In `wordpress__components`, I have just changed the `downshift` version, as its latest version was causing build failures on not only this, but [other PRs](54837) too.

File: types/wordpress__components/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "private": true,
     "dependencies": {
         "@wordpress/element": "^3.0.0",
-        "downshift": "^6.0.15",
+        "downshift": "6.0.15",
```
Comment (eps1lon) [inline types/wordpress__components/package.json:5]: Do we need this change here?

Comment (brc-dd) [reply types/wordpress__components/package.json:5]: Actually `downshift`'s latest version is causing build failures. [Here are the logs of the CI run.](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/938f33799d3b79ebec3bd5d4dba4c2c258096b2d/checks?check_suite_id=3408195176)

Comment (eps1lon) [reply types/wordpress__components/package.json:5]: Alright, then let's revert it here and wait for `master` to get fixed.

**Review by eps1lon — APPROVED** [review]

**Comment by brc-dd** [comment]
@eps1lon Hey, can you re-review the changes? The bot was stuck last time on The CI Failed and Needs Author Action, even after the CI checks have passed.

**Review by eps1lon — APPROVED** [review]

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by brc-dd** [comment]
Ready to merge

**Pull request merged**